feat: add third party services page

lcsfelix opened the pull request.

lcsfelix commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create 3rd party services page](https://trello.com/c/SJAJeejV/127-create-3rd-party-services-page)

RafaelTaranto commented on new-lamassu-admin/package.json:
```diff
@@ -32,7 +32,7 @@
     "@storybook/react": "^5.2.3",
     "babel-loader": "8.0.6",
-    "react-scripts": "3.2.0",
+    "react-scripts": "^3.2.0",
```
no reason to change this, undo package.json and package-lock.json changes.

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/formik/TextInput.js:
```diff
@@ -19,4 +49,80 @@ const TextInputFormik = memo(({ ...props }) => {
   )
 })
 
-export default TextInputFormik
```
this is a breaking change for screens using this component, you should create new components for PhoneNumberInputFormik, SecretInputFormik

RafaelTaranto commented on new-lamassu-admin/src/components/single-row-table/SingleRowTable.js:
```diff
@@ -0,0 +1,152 @@
+const { label1, p } = typographyStyles
+
+/* content = { title, category, items: [{ label, value }]}
+ */
```
remove this

lcsfelix replied:
Why? It's a bit complex, I'd think it's a good idea to leave some sort of documentation.

RafaelTaranto replied:
Oh, looks like a commented out piece of code. 

Why is all of this wrapped on the object? Might as well pass them as separate props. I also don't see category being used

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/Infura.js:
```diff
@@ -0,0 +1,119 @@
+    }),
+    handleSubmit: (values, actions) => {
+      console.log(values)
+    }
```
Handle submit should make a call to post the new config, it should be the same for all forms on this page.

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/Services.js:
```diff
@@ -0,0 +1,233 @@
+    formLabel: {
+      extend: label1
+    }
+  }
```
This looks big enough to get its own file, I'll write a small style guide for us.

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/EditService.js:
```diff
@@ -0,0 +1,119 @@
+    options: {
+      data: submission
+    }
+  })
```
This gets triggered at the wrong time, we should make the call explicit and only call on the form handle submit.

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/Twilio.js:
```diff
@@ -0,0 +1,138 @@
+      name: schema.toNumber.code,
+      label: schema.toNumber.display,
+      type: 'text',
+      component: PhoneNumberInputFormik
```
I don't think we should force them to write a number here, there's a bug fix open about how twilio accepts named phone numbers.

lcsfelix replied:
Should the input still detect a phone number and apply masking or should it just be a simple `TextInputFormik`?

RafaelTaranto replied:
I think it should just be a TextInputFormik. Bringing that up in the design chat.

RafaelTaranto committed:
fix: add cors to register endpoint

RafaelTaranto commented on lib/new-settings-loader.js:
```diff
@@ -11,7 +11,7 @@ low(adapter).then(it => {
 function saveConfig (config) {
   const currentState = db.getState()
-  const newState = _.merge(currentState, config)
+  const newState = _.assign(currentState, config)
```
Why the change here?

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/formik/TextInput.styles.js:
```diff
@@ -0,0 +1,28 @@
+import { fontColor, offColor } from '../../../styling/variables'
+import typographyStyles from '../../typography/styles'
```
We have absolute imports now, so whenever it's outside of the current folder you should go with it.

`src/styling/variables` instead of `../../../styling/variables`

same for the other one.

RafaelTaranto commented on new-lamassu-admin/src/components/single-row-table/SingleRowTable.js:
```diff
@@ -0,0 +1,160 @@
+  tableNewDisabledHeaderColor,
+  secondaryColorDarker
+} from '../../styling/variables'
+import typographyStyles from '../typography/styles'
```
Make them absolute.

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/Bitgo.js:
```diff
@@ -0,0 +1,250 @@
+import * as Yup from 'yup'
+
+import TextInputFormik from '../../components/inputs/formik/TextInput'
+import SecretInputFormik from '../../components/inputs/formik/SecretInput'
```
Same as above

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/aux.js:
```diff
@@ -0,0 +1,40 @@
+const formatLong = value =>
+  value
+    ? `${value.slice(0, 8)}(...)${value.slice(value.length - 8, value.length)}`
+    : ''
```
I don't think this works. 

test case: `smallword`

response: `smallwor(...)mallword`

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/EditService.js:
```diff
@@ -0,0 +1,104 @@
+              return {
+                code: p[0],
+                value: p[1]
+              }
```
Any reason to use this format instead of plain `{ key: value }`?

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/Services.js:
```diff
@@ -0,0 +1,245 @@
+    onCompleted: data => setAccountsConfig(data.saveConfig.accounts)
+  })
+
+  const useStyles = makeStyles(styles)
```
This should be outside of the component function.

RafaelTaranto commented on new-lamassu-admin/src/pages/Services/Services.js:
```diff
@@ -0,0 +1,245 @@
+      [it],
+      accountsConfig
+    )
+    return saveConfig({ variables: { config: { accounts: newAccounts } } })
```
I don't get this one, why the complex structure?

RafaelTaranto reviewed (approved):
lgtm

lcsfelix committed:
feat: add services page

fix: change styles, fix hook trigger, add ux

feat: setup custom error messages

refactor: conform to new style guide

refactor: migrate to graphql

refactor: migrate to Ramda

fix: update state on mutation

refactor: migrate error ux to graphql

fix: change structure of accounts config

fix: use absolute imports

fix: move makeStyles out of components

fix: correct Strike behaviour
